front/src/applications/operationalStudies/components/MacroEditor/ngeToOsrd.ts

```diff
@@ -147,16 +147,20 @@ const handleTrainrunOperation = async ({
   addDeletedTrainIds: (trainIds: number[]) => void;
 }) => {
   const { nodes } = netzgrafikDto;
-  const trainrunSectionsByTrainrunId = getTrainrunSectionsByTrainrunId(netzgrafikDto, trainrun.id);
+
   switch (type) {
     case 'create': {
+      const trainrunSectionsByTrainrunId = getTrainrunSectionsByTrainrunId(
+        netzgrafikDto,
+        trainrun.id
+      );
       const newTrainSchedules = await dispatch(
         osrdEditoastApi.endpoints.postV2TimetableByIdTrainSchedule.initiate({
           id: timeTableId,
           body: [
             {
               ...DEFAULT_PAYLOAD,
-              ...createTrainSchedulePayload(trainrunSectionsByTrainrunId, nodes, trainrun),
+              ...createTrainSchedulePayload(trainrunSectionsByTrainrunId!, nodes, trainrun),
```

Comment:
Do we really need non-null assertion here ?

Reply:
Oups done, thanks